plugin/hotswap-agent-mybatis-plugin/src/main/java/org/hotswap/agent/plugin/mybatis/transformers/MyBatisTransformers.java

```diff
@@ -183,4 +185,59 @@ public static void patchSqlSessionFactoryBean(CtClass ctClass, ClassPool classPo
         ctClass.addMethod(proxyMethod);
         LOGGER.debug("org.mybatis.spring.SqlSessionFactoryBean patched.");
     }
+
+    @OnClassLoadEvent(classNameRegexp = "org.mybatis.spring.mapper.MapperScannerConfigurer")
+    public static void patchMapperScannerConfigurer(CtClass ctClass, ClassPool classPool)
+            throws NotFoundException, CannotCompileException {
+        /*
+         * In org.mybatis.spring.mapper.MapperScannerConfigurer#processPropertyPlaceHolders,
+         * a BeanFactory is created using new DefaultListableBeanFactory() that contains only
+         * this mapper scanner and processes the factory. This is not needed;
+         * it should be removed from SpringChangedAgent after processing.
+         */
+        CtMethod processPropertyPlaceHolderM = ctClass.getDeclaredMethod("processPropertyPlaceHolders");
+        int lineNumber = processPropertyPlaceHolderM.getMethodInfo().getLineNumber(0);
+        int lineNum = lineNumber + 16;
+
+        // remove from SpringChangedAgent
+        String codeToInsert = "org.hotswap.agent.plugin.spring.reload.SpringChangedAgent.destroyBeanFactory(factory);";
+        processPropertyPlaceHolderM.insertAt(lineNum + 1, codeToInsert);
```

Comment:
Guess to insert at the end of the method instead in the middle. It would be invariant to possible method chenges in new versiosn

Reply:
I had a similar idea before. However, this is an intermediate variable within the `processPropertyPlaceHolders` method body, and I haven't found a way to insert code at the very beginning or end of this method to remove the caching logic. If you have any ideas, you can help me out. I looked at the history of the `org.mybatis.spring.mapper.MapperScannerConfigurer` class, and this `processPropertyPlaceHolders` method indeed hasn't changed much.

Reply:
I would guess to use ExprEditor (look to HA project usages) as follows (not sure if it is working):

``` java
            processPropertyPlaceHolderM.instrument(
                    new ExprEditor() {
                        public void edit(ConstructorCall c) throws CannotCompileException {
                            if (c.getClassName().equals(..))
                                m.replace("{ $_ = $proceed(); SpringChangedAgent.destroyBeanFactory($_); }");
                        }
                    });
```

Reply:
> I would guess to use ExprEditor (look to HA project for usages) as follows (not sure if it is working):
> 
> ```java
>             processPropertyPlaceHolderM.instrument(
>                     new ExprEditor() {
>                         public void edit(ConstructorCall c) throws CannotCompileException {
>                             if (c.getClassName().equals(..))
>                                 m.replace("{ $_ = $proceed(); SpringChangedAgent.destroyBeanFactory($_); }");
>                         }
>                     });
> ```
@skybber 
Thank you very much. I have followed the instructions to fix the code and ensure that changes to the processPropertyPlaceHolders method body do not affect the execution of the MyBatis plugin instrumentation code

Reply:
![image](https://github.com/user-attachments/assets/cbdeaff9-3751-488b-b79e-6c2ffbf58e5a)